[TC-8713] [TC-9481] [TC-9467] Explain how to handle open requests using polling (#83)
tradecloud/tradecloud-docs-api-v2: order/supplier/receive/simple-order-event.md

---
description: How to receive a simple order sent by the buyer.
---

> # Receive an order

Reviewer: Note: somehow this is not the header that is rendered on this page..

Author: I see this
![image](https://github.com/tradecloud/tradecloud-docs-api-v2/assets/1092388/e86fee01-5519-40d9-bb45-ba389fbcf0b0)

Reviewer: ![Screenshot 2024-03-05 at 14 49 34](https://github.com/tradecloud/tradecloud-docs-api-v2/assets/46396634/e5034a29-d838-481d-b8a8-af452f000424)

Are we looking at the same page? I'm looking at `processes/order/supplier/receive`. It seems your screenshot if of `processes/order/supplier/README.md`?

Author: The Gitbook uses the title from the glossary

Reviewer: Alright, I've pushed a small change to make it consistent with the summary. Just to avoid confusion

> Tradecloud will send a purchase order, either new or updated, to the supplier when an order event has been triggered.

Reviewer: Only if the supplier is integrated with the webhook, right? I think this can use some elaboration.

Author: Improved

Reviewer: I don't see the change?

Author: See the additional lines below and https://app.gitbook.com/o/-LNyIUZSvpZWZ81yEr7V/s/-LNyJ9UuwLCjTvA2sqQR-887967055/~/revisions/nZCqdUTDMvvrpj07rcDN/processes/order/supplier/receive/simple-order-event

![image](https://github.com/tradecloud/tradecloud-docs-api-v2/assets/1092388/827d9b36-9281-42f6-a82f-9d75855f1653)

## `simpleOrderEvent`

> This page assumes you are using the webhook with the simple delivery schedule in the `simpleOrderEvent`.

Reviewer: I'd move this up for clarification.
Also; what if this is not the case?

```suggestion
This page assumes the supplier integrated by using the webhook, using the simple delivery schedule in the `simpleOrderEvent`.
```

Author: Moved up and added a back link to the README

> * `orderId`: the Tradecloud order identifier.

Reviewer: Seeing this long list of all properties, written out to the deepest level; are we sure we don't want to improve the API docs instead? 
I think it will increase maintainability a lot if we only mention a high-level summary and only the functional considerations in a page like this..

Author: I have added this to https://tradecloud.atlassian.net/browse/TC-9548

* `supplierAccountNumber`: your account number as known in the buyer's ERP system.
* `description`: a free format additional description of this order added by the buyer.
* `contact`: the buyer employee responsible for this order.
* `properties`: are key-value based custom fields, added by the buyer.
* `notes`: are simple custom fields, added by the buyer.
* `labels`: value-added services labels on order level.
* `documents`: contain meta data, objectId or url, of attached documents by the buyer.
* `orderType`: the order type, one of `Purchase` or `Forecast`.

{% page-ref page="download-document.md" %}

### Supplier order

`supplierOrder` is mostly an echo of your order fields as explained in [Send order response](../send-order-response/)​.

* `buyerAccountNumber`: the buyer account number as known in your ERP system.

{% hint style="warning" %}
The `buyerAccountNumber` should be set on forehand in the Tradecloud connection with your buyer. You can set the account code when inviting a new connection or in the connection overview in the portal.
{% endhint %}

### Order status

The order status is the aggregation of all the lines statuses.

#### Order process status

> {% hint style="info" %}

Reviewer: Why are these hint blocks? 
imo it's best to only use these when a part of the text is more important than the rest.

Author: To make a distinction between fields and values.
![image](https://github.com/tradecloud/tradecloud-docs-api-v2/assets/1092388/236990a8-6e05-4b60-9280-04a7222e3828)

Reviewer: Ah I see. Ok, that works.

The order **process** status is one of:

* `Issued`: the order is \(re\)issued by the buyer.
* `InProgress`: the order is under negotiation between buyer and supplier
* `Confirmed`: the order is completely agreed between buyer and supplier
* `Rejected`: the order is completely rejected by supplier
* `Completed`: the order is completed at the buyer
* `Cancelled`: the order is cancelled by the buyer
{% endhint %}

#### Order logistics status

> {% hint style="info" %}

Reviewer: Nit: convert to regular text

Author: To make a distinction between fields and values.

The order **logistics** status is one of:

* `Open`: no or partial quantity Produced, ReadyToShip, Shipped or Delivered
* `Produced`: the order full quantity is produced by the supplier
* `ReadyToShip`: the order full quantity is ready to be shipped by the supplier
* `Shipped`: the order full quantity is shipped by the supplier
* `Delivered`: the order full quantity is delivered at the buyer
* `Cancelled`: the order is cancelled by the buyer
{% endhint %}

> ## Order lines

Reviewer: This is still part of the `## simpleOrderEvent`
```suggestion
### Order lines
```

Reviewer: Headers below may need to be adjusted accordingly.
Or perhaps use: 
- `## simpleOrderEvent header`
- `## simpleOrderEvent lines`

Author: Updated, but we did not call it `header` in the API.
